src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -345,7 +345,7 @@
 
 (defmethod set-cell! String
   [^Cell cell value _styles _typed-styles]
-  (.setCellValue cell ^String value))
+  (.setCellValue cell ^String (if (> (count value) 32767) (subs value 0 32767) value)))
```

Comment:
We have a helper function `metabase.util/truncate` that can be used here instead

Reply:
The limit value would probably also be worth pulling out into a standalone var. You can mark it as `*dynamic*` to make it easier to test (by re-binding it to a smaller value in the test). You can see an example with the existing `*auto-sizing-threshold*` constant.

Reply:
I think it's also worth clarifying if 32767 applies to number of characters or number of bytes

Reply:
> I think it's also worth clarifying if 32767 applies to number of characters or number of bytes

It should be the number of characters. In what cases would it be the number of bytes?
https://support.microsoft.com/en-us/office/excel-specifications-and-limits-1672b34d-7043-467e-8e27-269d656771c3